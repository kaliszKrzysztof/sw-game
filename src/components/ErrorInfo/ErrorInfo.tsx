import React from 'react';
import Image from 'next/image';
import { Typography } from '@material-ui/core';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      margin: theme.spacing(8, 0),
    },
    title: {
      marginTop: theme.spacing(4),
    },
  }),
);

interface ErrorInfoProps {
  title: string;
  message: string;
}

const ErrorInfo: React.FC<ErrorInfoProps> = ({ title, message }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Image
        src="/darth_vader.png"
        alt="Something went wrong"
        width={293}
        height={300}
        data-testid="image"
      />
      <Typography
        className={classes.title}
        align="center"
        color="textSecondary"
        variant="h4"
        gutterBottom
        data-testid="title"
      >
        {title}
      </Typography>
      <Typography align="center" color="textSecondary" data-testid="message">
        {message}
      </Typography>
    </div>
  );
};

export default ErrorInfo;
