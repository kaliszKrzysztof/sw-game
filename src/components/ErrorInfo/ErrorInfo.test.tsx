import React from 'react';
import { render } from '@testing-library/react';
import ErrorInfo from './ErrorInfo';

describe('<ErrorInfo />', () => {
  it('should render', () => {
    render(<ErrorInfo title="Test" message="Test" />);
  });
  it('matches the snapshot', () => {
    const { container } = render(<ErrorInfo title="Test" message="Test" />);
    expect(container).toMatchSnapshot();
  });
  it('should render correct items', () => {
    const { getByTestId } = render(
      <ErrorInfo title="Title" message="Message" />,
    );
    expect(getByTestId('title').textContent).toEqual('Title');
    expect(getByTestId('message').textContent).toEqual('Message');
    expect(getByTestId('image')).toBeInTheDocument();
  });
});
