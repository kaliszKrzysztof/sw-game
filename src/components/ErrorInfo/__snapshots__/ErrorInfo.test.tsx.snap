// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorInfo /> matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-3"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjkzIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="Something went wrong"
        data-testid="image"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <h4
      class="MuiTypography-root makeStyles-title-4 MuiTypography-h4 MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-alignCenter"
      data-testid="title"
    >
      Test
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      data-testid="message"
    >
      Test
    </p>
  </div>
</div>
`;
